src/components/TodoItem.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  handleDeletingTodo?: (id: number) => void,
+  tempTodo?: Todo | null
+  todosForTemp: number[]
+  handleUpdatingTodo: (id: number, data: null | string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleDeletingTodo,
+  tempTodo,
+  todosForTemp,
+  handleUpdatingTodo,
+}) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const [titleValue, setTitleValue] = useState(title);
+  const [isChanged, setIsChanged] = useState(false);
+
+  const handleDoubleClick = () => {
+    setIsChanged(true);
+  };
+
+  const handlerCancelEditing = () => {
+    setIsChanged(false);
+    setTitleValue(title);
+  };
+
+  const handleSaveChanges = () => {
+    setIsChanged(false);
+    if (!titleValue.length) {
+      handleDeletingTodo?.(id);
```

Comment:
remove optional chaining

Reply:
it was needed, because of Liinter ((
if I remove optional chaining, I have to add it to the condition of "if"

Reply:
I will remove optional chaining.